docs/conf.py

```diff
@@ -42,6 +42,27 @@
 # This pattern also affects html_static_path and html_extra_path.
 exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
 
+nitpick_ignore = [
+    ("py:class", "optional"),
+    ("py:class", "abc.ABC"),
+]
+
+nitpick_ignore_regex = [
+    ("py:class", r"algosdk\.abi\..*"),
+    ("py:class", r"contextlib\..*"),
+    ("py:class", r"enum\..*"),
+    ("py.obj", r".*\.T$"),
+    ("py.obj", r".*\.T[0-9]+$"),
+    ("py.class", r".*\.T_co$"),
+    ("py.obj", r".*\.T_co$"),
+    ("py.class", r".*\.T$"),
+    ("py.class", r"^T$"),
+    ("py.obj", r".*\.N$"),
+    ("py.class", r".*\.N$"),
+    ("py.class", r"^N$"),
+    ("py.class", r"^type\[.*"),
+    ("py.obj", r"typing\..*"),
+]
```

Comment:
Could you explain how this was generated and what it does?

Reply:
These are sphinx trying to index from `abi` type arguments, and the erroring comes from `-n` nitpicky-mode.

This config asks to ignore the error coming from `py:class` or `py:obj` cannot find reference for something matching the regex.

Reply:
after the discussion we decided to close this thread, for we are removing the nitignore configs in `conf.py`.